### PR TITLE
Remove pytorch-pretrained-bert

### LINKED ISSUE #3067: upgrade pytorch-pretrained-bert => pytorch-transformers
they renamed the library, I'm not sure if they changed any APIs

### LINKED ISSUE #3481: Dependencies out-of-date: pytorch-pretrained-bert, pytorch-transformers, BertAdam, conllu, overrides
These dependencies are all out-of-date:

* pytorch-pretrained-bert* → transformers
* pytorch-transformers* → transformers
* BertAdam → AdamW
* conllu== → conllu>=
* overrides== → overrides>=

For reference, see code edits at this MacPorts PR, https://github.com/macports/macports-ports/pull/5862

https://github.com/macports/macports-ports/blob/25bcfba5003b27ad91f3e3bc1716162496686988/python/py-allennlp/Portfile#L74-L102

### PR DESCRIPTION
**Pull request by DeNeutoy**
Fixes #3067 , fixes #3481

### CODE REVIEW

File: allennlp/modules/token_embedders/bert_token_embedder.py [hunk 1]
```diff
@@ -186,11 +188,12 @@ def forward(
 
         # input_ids may have extra dimensions, so we reshape down to 2-d
         # before calling the BERT model and then reshape back at the end.
-        all_encoder_layers, _ = self.bert_model(
+        _, _, all_encoder_layers = self.bert_model(
             input_ids=util.combine_initial_dims(input_ids),
             token_type_ids=util.combine_initial_dims(token_type_ids),
             attention_mask=util.combine_initial_dims(input_mask),
         )
+        print([x.shape for x in all_encoder_layers])
```
Comment (matt-gardner) [inline allennlp/modules/token_embedders/bert_token_embedder.py]: Stray print

File: allennlp/tests/models/bert_srl_test.py [hunk 2]
```diff
@@ -16,7 +16,8 @@ def setUp(self):
         # monkeypatch the PretrainedBertModel to return the tiny test fixture model
         config_path = self.FIXTURES_ROOT / "bert" / "config.json"
         vocab_path = self.FIXTURES_ROOT / "bert" / "vocab.txt"
-        config = BertConfig(str(config_path))
+        config = BertConfig.from_json_file(config_path)
+        print(config)
```
Comment (matt-gardner) [inline allennlp/tests/models/bert_srl_test.py]: Stray print

File: allennlp/training/optimizers.py [hunk 3]
```diff
@@ -158,7 +157,6 @@ def from_params(cls, model_parameters: List, params: Params):  # type: ignore
     "rmsprop": (torch.optim.RMSprop, None),
     "adamax": (torch.optim.Adamax, None),
     "averaged_sgd": (torch.optim.ASGD, None),
-    "bert_adam": (BertAdam, None),
```
Comment (matt-gardner) [inline allennlp/training/optimizers.py]: @mhagiwara, FYI, looks like we should use `huggingface_adamW` instead of `bert_adam`.

Comment (DeNeutoy) [reply allennlp/training/optimizers.py]: Do you know when this was added? I would like to change it to just `adamw`

Comment (matt-gardner) [reply allennlp/training/optimizers.py]: I don't know, but if `adamw` is a relatively standard / reasonable name, I'm totally fine with renaming it.  Also not hard to register it twice.

Comment (matt-gardner) [reply allennlp/training/optimizers.py]: Oh, at that rate, should also be fine to just have `bert_adam` use the new one, also, so we don't break anyone.

Comment (DeNeutoy) [reply allennlp/training/optimizers.py]: It takes different arguments though

**Comment by DeNeutoy** [comment]
@matt-gardner just double checking this looks ok to you - I had to add a temp fix for a bug in huggingface which is a bit hacky. Also because of #3637 we have a failing test, but it's unrelated.

**Comment by matt-gardner** [comment]
Sorry, wasn't sure what you wanted me to answer in the slack channel, and I was trying to get this `Lazy[]` stuff actually working...

**Comment by DeNeutoy** [comment]
Yeah no problem it was more just a shout into the void

**Comment by matt-gardner** [comment]
Except this had a failing test on CI - I didn't mean to override that.  That blocks all future PRs until the test is fixed.  If it's the drop connect issue, it seems fine to me to disable that test.

File: allennlp/training/optimizers.py (hunk 3 above)
Comment (mhagiwara) [reply allennlp/training/optimizers.py]: @matt-gardner thanks for the info!

**Pull request merged**